Alex and Happy Part 2/article.md

```diff
@@ -0,0 +1,1381 @@
+```
+
+Or, simplifying a bit:
+
+```haskell
+Right
+  [ Dec _ (Name _ "the_answer") [] (Just (TVar _ (Name _ "int")))
+    (ELetIn _ (Dec _ (Name _ "a") [] Nothing (EInt _ 20))
+    (ELetIn _ (Dec _ (Name _ "b") [] Nothing (EInt _ 1))
+    (ELetIn _ (Dec _ (Name _ "c") [] Nothing (EInt _ 2))
+    (EBinOp _ (EBinOp _ (EVar _ (Name _ "a")) (Times _) (EVar _ (Name _ "c"))) (Plus _) (EBinOp _ (EVar _ (Name _ "b")) (Times _) (EVar _ (Name _ "c")))))))
+  , Dec _ (Name _ "main") [Argument _ (Name _ "unit") (Just (TUnit _))] (Just (TUnit _))
+    (EApp _ (EVar _ (Name _ "print")) (EPar _ (EBinOp _ (EString _ "\"The answer is: \"") (Plus _) (EVar _ (Name _ "the_answer")))))
+  ]
+```
+
+To finish, let's just make one last change to our parser. At the top of the file, add the following:
+
+```diff
+ %lexer { lexer } { L.RangedToken L.EOF _ }
++%expect 0
+```
+
+This tells Happy that it should expect exactly zero shift/reduce conflicts.
+Happy will always expect no reduce/reduce conflicts when this directive is present, no matter the provided value.
+
+You can find the complete grammar [here](https://gist.github.com/heitor-lassarote/5f24f40c8625f25c6108ee59dc10e6d5).
+
+### Exercises
+
+1. Change your lexer and parser to accept fractional numbers.
+Besides accepting numbers such as `3.14`, it should also accept exponents, like `12.5e-4`.
+
+<details>
```

Comment:
Hint jumps out of the list element. 
![image](https://user-images.githubusercontent.com/35335196/173023730-2563dfbd-14fa-4a9c-81c1-865524272da3.png)

Reply:
In my markdown viewer, I put `<br>` because everything was glued together, but I removed it if it'll look better in the blog.

Reply:
I believe to avoid the hint "jumping out" you need to indent the whole `<details>` block. We use markdown-it on the blog, so it will parse/render similarly to https://markdown-it.github.io/

Reply:
They seem to render differently for me, and indenting details doesn't seem to help... to be honest, adding `<br>` also looks better in this link, so I left this block of code as it is. @NaeosPsy, is it better in the blog without this? Or should I still try to figure out how to make it prettier?

Reply:
I think it should be indented. 

I am having problems with it myself, haha, but you can check out @lierdakil's article or something, it has that.